common4j/src/main/com/microsoft/identity/common/java/controllers/ExceptionAdapter.java

```diff
@@ -471,4 +480,24 @@ private static boolean isIntunePolicyRequiredError(
                 errorResponse.getError().equalsIgnoreCase(OAuth2ErrorCode.UNAUTHORIZED_CLIENT) &&
                 errorResponse.getSubError().equalsIgnoreCase(OAuth2SubErrorCode.PROTECTION_POLICY_REQUIRED);
     }
+
+    /**
+     * Identifies whether an error is specific to native authentication MFA scenarios.
+     * @param errorResponse
+     * @return true if errorReponse is a native authentication MFA error
+     */
+    private static boolean isNativeAuthenticationMFAError(
+            @NonNull final TokenErrorResponse errorResponse) {
+        if (!(errorResponse instanceof MicrosoftTokenErrorResponse)) {
+            return false;
+        }
+
+        MicrosoftTokenErrorResponse microsoftTokenErrorResponse = ((MicrosoftTokenErrorResponse) errorResponse);
+        return !StringUtil.isNullOrEmpty(microsoftTokenErrorResponse.getError()) &&
+                !StringUtil.isNullOrEmpty(microsoftTokenErrorResponse.getSubError()) &&
+                !microsoftTokenErrorResponse.getErrorCodes().isEmpty() &&
+                errorResponse.getError().equalsIgnoreCase(OAuth2ErrorCode.INVALID_GRANT) &&
+                errorResponse.getSubError().equalsIgnoreCase(OAuth2SubErrorCode.BASIC_ACTION) &&
+                microsoftTokenErrorResponse.getErrorCodes().contains((long) 50076);
```

Comment:
In iOS we check only that the first error code is 50076. We don't check error or suberror field

Reply:
Updated